### PR TITLE
Checks on Patch Spawning

### PR DESCRIPTION
**Pull request by rgknox**
### Description:

This changeset adds some checks during the patch spawning routine that ensure new patches are only created, and new areas to those patches are only donated, if those areas are reasonably non-zero.  

In long-term tests on cheyenne some near-zero (maybe zero) area patches were being formed, and crashing the runs. This fix prevents those crashes and numerical overflows associated with them.

Also, the lower bounds for acceptable patch sizes were tweaked to more reasonable levels.

### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [ ] If answers were expected to change, evaluation was performed and provided

### Test Results:

These changes allowed a 100 year, 14pft simulation on master to complete successfully.

Standard regression tests TBD

CTSM (or) E3SM (specify which) test hash-tag:

CTSM (or) E3SM (specify which) baseline hash-tag:

FATES baseline hash-tag:

Test Output:

### CODE REVIEW

**Comment by jkshuman** [comment]
Thanks for the heads up on this @rgknox. I am surprised I hadn't crashed this in my runs. Happy to test this across SA with my current parameter set.

**Comment by rgknox** [comment]
Regression tested on cheyenne intel, fates, all PASS:
/gpfs/fs1/scratch/rgknox/clmed-tests/fates.cheyenne.intel.spawn-patch-cfix-C5dc7f0a-F0990f54

File: biogeochem/EDPatchDynamicsMod.F90 [hunk 1]
```diff
@@ -301,6 +301,7 @@ subroutine spawn_patches( currentSite, bc_in)
     
     use EDParamsMod         , only : ED_val_understorey_death, logging_coll_under_frac
     use EDCohortDynamicsMod , only : zero_cohort, copy_cohort, terminate_cohorts 
+    use ChecksBalancesMod   , only : SiteCarbonStock
```
Comment (ckoven) [inline biogeochem/EDPatchDynamicsMod.F90]: seems like this isn't needed since all calls to it have been commented out?

Comment (rgknox) [reply biogeochem/EDPatchDynamicsMod.F90]: yeah.  Maybe I should remove the comments too.  I suppose there is no reason to keep them. Will update

**Comment by rgknox** [comment]
all pass, cheyenne, intel, fates test suite:
/gpfs/fs1/scratch/rgknox/clmed-tests/fates.cheyenne.intel.spawn-cfix-C5dc7f0a-Fa6ac612

**Pull request merged**